compress.cc

```diff
@@ -22,7 +22,7 @@ class lz4_processor: public compressor {
     size_t uncompress(const char* input, size_t input_len, char* output,
                     size_t output_len) const override;
     size_t compress(const char* input, size_t input_len, char* output,
-                    size_t output_len) const override;
+                    size_t output_len) override;
```

Comment:
I think compress() is still logically const, since it doesn't change the compressor, only the output.

This is one of the few cases where `mutable` is indicated. It's state that doesn't affect the result of computations.